peacebot/core/plugins/Moderation/mod.py

```diff
@@ -0,0 +1,259 @@
+    higher_role_check(ctx.member, member)
+    mod_logs = await mod_logs_check(ctx)
+    now = datetime.now(timezone.utc)
+    await member.edit(communication_disabled_until=now, reason=ctx.options.reason)
+    embed = (
+        hikari.Embed(
+            description=f"🔊 Removed TimeOut -> {member}\n**Reason:** {ctx.options.reason}",
+            color=EmbedColors.SUCCESS,
+            timestamp=datetime.now().astimezone(),
+        )
+        .set_author(
+            name=f"{ctx.author}(ID {ctx.author.id})", icon=ctx.author.avatar_url
+        )
+        .set_thumbnail(member.avatar_url)
+    )
+    await mod_logs.send(embed=embed)
+    response = await ctx.respond(embed=embed)
+    message_link = (await response.message()).make_link(ctx.guild_id)
+    await register_cases(
+        guild_id=ctx.guild_id,
+        moderator=ctx.author.id,
+        target=member.id,
+        reason=ctx.options.reason,
+        message_link=message_link,
+        channel_id=ctx.channel_id,
+        type="TimeOut Disable",
+    )
+
+
+@mod_plugin.command
+@lightbulb.option("reason", "The reason to kick the member")
+@lightbulb.option("member", "The member to kick", type=hikari.Member)
+@lightbulb.command("kick", "Kick the member from the server")
+@lightbulb.implements(lightbulb.SlashCommand, lightbulb.PrefixCommand)
+@mod_role_check
+async def kick_command(ctx: lightbulb.Context) -> None:
+    member: hikari.InteractionMember = ctx.options.member
+    higher_role_check(ctx.member, member)
+    mod_logs = await mod_logs_check(ctx)
+    await member.kick(reason=ctx.options.reason)
+    embed = (
+        hikari.Embed(
+            description=f"👢 Kicked -> {member}\n**Reason:** {ctx.options.reason}",
+            color=EmbedColors.INFO,
+            timestamp=datetime.now().astimezone(),
+        )
+        .set_author(
+            name=f"{ctx.author}(ID {ctx.author.id})", icon=ctx.author.avatar_url
+        )
+        .set_thumbnail(member.avatar_url)
+    )
+    await mod_logs.send(embed=embed)
+    response = await ctx.respond(embed=embed)
+    message_link = (await response.message()).make_link(ctx.guild_id)
+    await register_cases(
+        guild_id=ctx.guild_id,
+        moderator=ctx.author.id,
+        target=member.id,
+        reason=ctx.options.reason,
+        message_link=message_link,
+        channel_id=ctx.channel_id,
+        type="Kick",
+    )
+
+
+@mod_plugin.command
+@lightbulb.option(
+    "days",
+    "Specify number of days of messages to delete(Min - 0, Max- 7)",
+    required=False,
+    type=int,
+)
+@lightbulb.option("reason", "The reason to ban the member")
+@lightbulb.option("member", "The member to ban", type=hikari.Member)
+@lightbulb.command("ban", "Ban the member from the server")
+@lightbulb.implements(lightbulb.SlashCommand, lightbulb.PrefixCommand)
+@mod_role_check
+async def ban_command(ctx: lightbulb.Context) -> None:
+    member: hikari.InteractionMember = ctx.options.member
+    higher_role_check(ctx.member, member)
```

Comment:
It'd be better if you add the check using some functionality to add checks in Hikari, rather than doing this inside the function body. 

If you're sure you want to go this way, you could rather return `bool` from the function. And handle that bool accordingly. Or create a decorator, as this might be used in various other places.

Reply:
Creating a decorator wouldn't really make  sense as we have to pass the author and the target of the command.
I will see if I can make this a lightbulb check rather than calling this function.

Reply:
Sure goodluck.